infra/baseServer_test.go

```diff
@@ -160,10 +177,114 @@ func TestAgentAgentMessage(t *testing.T) {
 	}
 }
 
-func TestAgentListeningSpinner(t *testing.T) {
+func TestAgentListeningSpinnerOpen(t *testing.T) {
+	const numAgents = 3
+	m := make([]infra.AgentGeneratorCountPair[ITestBaseAgent], 1)
+	m[0] = infra.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
+	server := infra.CreateServer[ITestBaseAgent](m, 1, time.Second, 2)
+	server.Initialise()
+	msg := NewTestMessage()
+	agentMap := server.GetAgentMap()
+	arrayOfIDs := make([]uuid.UUID, numAgents)
+	i := 0
+	for id := range agentMap {
+		arrayOfIDs[i] = id
+		i++
+	}
+	server.BeginAgentListeningSession()
+	server.SendMessage(msg, arrayOfIDs)
+	fmt.Println(arrayOfIDs)
+	//time.Sleep(10*time.Second)
+	server.EndAgentListeningSession()
+	for _, ag := range agentMap {
+		testAgent := ag.(*TestAgent)
+		if testAgent.receivedMessage == false {
+			t.Errorf("agent did not receive a message")
+		}
+	}
+}
 
+func TestAgentListeningSpinnerClosed(t *testing.T) {
+	const numAgents = 3
+	m := make([]infra.AgentGeneratorCountPair[ITestBaseAgent], 1)
+	m[0] = infra.MakeAgentGeneratorCountPair(NewTestAgent, numAgents)
+	server := infra.CreateServer[ITestBaseAgent](m, 1, time.Second, 2)
+	server.Initialise()
```

Comment:
there's a fair amount of repeated code here. could we add some kind of generator to the test file? I.e.:

`serv := getTestingServer(numAgent = 3)`

or whatever parameters you'll be needing. we can then avoid having to manually call `Initialise()`